src/browser/devtools.js

```diff
@@ -0,0 +1,121 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define((require, exports, module) => {
+
+  'use strict';
+
+  const {DOM} = require('react');
+  const Component = require('omniscient');
+  const {Record, List} = require('typed-immutable/index');
+
+  const DevTools = Record({
+    enableHUD: Boolean(false),
+    enableRemoteDevtools: Boolean(false),
+    console: Record({
+      // FIXME: I don't how to update these list from anywhere in our codebase.
+      log: List(String),
```

Comment:
Lists implement Array interface and you can use same `update` and `set` stuff as with other structures. For example you could do:

```js
const log = (devtools, message) =>
  devtools.set('console', devtools.console.push(message)
```